src/open_inwoner/openklant/clients.py

```diff
@@ -1,26 +1,246 @@
+        try:
+            response = self.get(url)
+            data = get_json_response(response)
+        except (RequestException, ClientError) as e:
+            logger.exception("exception while making request", exc_info=e)
+            return
+
+        contact_moment = factory(ContactMoment, data)
+
+        return contact_moment
+
+    def retrieve_objectcontactmomenten_for_contactmoment(
+        self, contactmoment: ContactMoment
+    ) -> List[ObjectContactMoment]:
+        try:
+            response = self.get(
+                "objectcontactmomenten", params={"contactmoment": contactmoment.url}
+            )
+            data = get_json_response(response)
+            all_data = list(pagination_helper(self, data))
+        except (RequestException, ClientError) as e:
+            logger.exception("exception while making request", exc_info=e)
+            return []
+
+        object_contact_momenten = factory(ObjectContactMoment, all_data)
+
+        # resolve linked resources
+        object_mapping = {}
+        for ocm in object_contact_momenten:
+            assert ocm.contactmoment == contactmoment.url
+            ocm.contactmoment = contactmoment
+            if ocm.object_type == "zaak":
+                object_url = ocm.object
+                # Avoid fetching the same object, if multiple relations wit the same object exist
+                if ocm.object in object_mapping:
+                    ocm.object = object_mapping[object_url]
+                else:
+                    ocm.object = fetch_case_by_url_no_cache(ocm.object)
+                    object_mapping[object_url] = ocm.object
+
+        return object_contact_momenten
+
+    def retrieve_klantcontactmomenten_for_klant(
+        self, klant: Klant
+    ) -> List[KlantContactMoment]:
+        try:
+            response = self.get(
+                "klantcontactmomenten",
+                params={"klant": klant.url},
+            )
+            data = get_json_response(response)
+            all_data = list(pagination_helper(self, data))
+        except (RequestException, ClientError) as e:
+            logger.exception("exception while making request", exc_info=e)
+            return []
+
+        klanten_contact_moments = factory(KlantContactMoment, all_data)
+
+        # resolve linked resources
+        for kcm in klanten_contact_moments:
+            assert kcm.klant == klant.url
+            kcm.klant = klant
+            kcm.contactmoment = self.retrieve_contactmoment(kcm.contactmoment)
+
+        return klanten_contact_moments
+
+    def retrieve_objectcontactmomenten_for_object_type(
+        self, contactmoment: ContactMoment, object_type: str
+    ) -> List[ObjectContactMoment]:
+
+        moments = self.retrieve_objectcontactmomenten_for_contactmoment(contactmoment)
+
+        # eSuite doesn't implement a `object_type` query parameter
+        ret = [moment for moment in moments if moment.object_type == object_type]
+
+        return ret
```

Comment:
1. Since we're not using the list (?), you could:

- `yield from (moment for moment in moments if...)`
- (in the following function): return next(ocms, None)

2. This seems to be the only use of the function which potentially returns multiple objects, but then we're only retrieving and returning the first (in `retrieve_objectcontactmoment`). Is this to allow for possible extensions where we actually use multiple `objectcontactmomenten` from this? Otherwise this and the following function could be collapsed.

Reply:
I could collapse it into one function `retrieve_objectcontactmoment`, since `retrieve_objectcontactmomenten_for_object_type` is indeed only used by that function currently. Though we might need this in the future, so that's why I added it initially